source/extensions/http/header_formatters/preserve_case/preserve_case_formatter.h

```diff
@@ -13,11 +14,17 @@ namespace PreserveCase {
 class PreserveCaseHeaderFormatter : public Envoy::Http::StatefulHeaderKeyFormatter {
 public:
   // Envoy::Http::StatefulHeaderKeyFormatter
+  PreserveCaseHeaderFormatter(const bool forward_reason_phrase);
+
   std::string format(absl::string_view key) const override;
   void processKey(absl::string_view key) override;
+  void setReasonPhrase(const char* data, uint32_t length) override;
+  absl::string_view getReasonPhrase() const override;
 
 private:
   StringUtil::CaseUnorderedSet original_header_keys_;
+  bool forward_reason_phrase_{false};
+  absl::string_view reason_phrase_;
```

Comment:
This needs to be std::string or you will be referencing data that is gone.

Reply:
Fixed.